### PR TITLE
Remove redundant code

### PR DESCRIPTION
**Pull request by shichanglin5**
image
this code is unreachable ?

### CODE REVIEW

**Comment by bboreham** [comment]
> this code is unreachable ?

I don't think so.  The example `l!="foo"` should use that code.

**Comment by shichanglin5** [comment]
> > this code is unreachable ?
> 
> I don't think so. The example `l!="foo"` should use that code.

image

For `l! = "foo" `, in the above code,`labelMustBeSet[m.Namw]= true` will not be executed, so the following switch case will go to the logic of default branch

**Comment by bboreham** [comment]
Consider `{l!="foo", l=~"."}`. 

There are two loops: the first sets `labelMustBeSet` based on all matchers, and the second considers each matcher in detail.

**Comment by shichanglin5** [comment]
> Consider `{l!="foo", l=~"."}`.
> 
> There are two loops: the first sets `labelMustBeSet` based on all matchers, and the second considers each matcher in detail.

Got it, thanks!

**Comment by bboreham** [comment]
The code is hard to follow here. 
If you would like to extend the comments to make it easier, please do.
